### ISSUE TITLE
Remote Debugger Hanging Up Crashes Byebug

### ISSUE THREAD

**Issue by LtSquigs** [deivid-rodriguez/byebug#274]
**Problem:** If the socket for a remote connection is closed after connecting to the remote server, further writes/reads will crash the server.

**Version:** 9.05

**Steps to Reproduce**:
 1) Run this example program

```
require 'byebug'
require 'byebug/core'

Byebug.wait_connection = true
Byebug.start_server 'localhost', ENV.fetch("BYEBUG_SERVER_PORT", 1048).to_i

puts 'a'
byebug
puts 'b'
sleep 5
puts 'c'
byebug
puts 'd'
```

2) Connect to this using the remote debugger
3) When it encounters the first `byebug` statement, `continue`
4) While the program sleeps, terminate the byebug CLI (CTRL + C)
5) After the program wakes up, when it hits the second `byebug` statement, it will crash with an `Errno::EPIPE` exception

**Expectation**:  The server should either detect when the socket is closed or if an EPIPE exception has happened, and stop communicating over the socket (treat it as if there was no connection).

While the above example is very arbitrary, a more real world example of where this happens: debugging remote rails applications. If you have a byebug statement in a controller, connect to it remotely, debug the area, then the debugger closes for some reason you will have a crash.

Fixing this will also allow users to "detach" the debugger by interrupting it between byebug calls (useful if you have a constantly running dev server).

**Comment by deivid-rodriguez** [comment]
@LtSquish Thanks for the great report. Want to try fixing it? 😄

**Comment by deivid-rodriguez** [comment]
@LtSquigs This should be fixed by #406, feel free to give the branch a try and give feedback!
